mlir/tools/mlir-tblgen/OpFormatGen.cpp

```diff
@@ -1374,7 +1381,14 @@ if (attr || /*isRequired=*/{1}) {{
 )decl",
                     namedAttr.name, isRequired);
   }
-  body << "return ::mlir::success();\n";
+  body << R"decl(
+for (NamedAttribute attr : dict) {
+  if (!usedKeys.contains(attr.getName()))
+    return emitError() << "Unknown key '" << attr.getName() <<
```

Comment:
s/Unknown/unknown/ (https://llvm.org/docs/CodingStandards.html#error-and-warning-messages)